packages/gatsby/src/query/file-parser.js

```diff
@@ -111,18 +111,20 @@ Also note that we are currently unable to use queries defined in files other tha
 async function parseToAst(filePath, fileStr, { parentSpan, addError } = {}) {
   let ast
 
+  const cleanFilePath = filePath.split(`?`)[0]
```

Comment:
We'll need to make this cleaner somehow as this doesn't make any sense without having ocntext of this project. (maybe start with adding a comment)